lib/assert.js

```diff
@@ -89,14 +89,25 @@ x.throws = function (fn, err, msg) {
 	}
 
 	try {
+		var errMsg;
 		if (typeof err === 'string') {
-			var errMsg = err;
+			errMsg = err;
 			err = function (err) {
 				return err.message === errMsg;
 			};
 		}
 
 		assert.throws(fn, err, msg);
+
+		if (errMsg) {
+			return Error(errMsg);
```

Comment:
Might actually be able to kill this if-block entirely given your suggestion at https://github.com/sindresorhus/ava/pull/576/files#r54448454. I'll give it a whirl!